macro-plantuml-macro/src/main/java/org/xwiki/contrib/plantuml/PlantUMLGenerator.java

```diff
@@ -46,6 +67,11 @@ public interface PlantUMLGenerator
      * @throws IOException when there's a generation or writing error
      * @since 2.4
      */
-    void outputImage(String input, OutputStream output, String serverURL, PlantUMLDiagramFormat format)
-            throws IOException;
+    default void outputImage(String input, OutputStream output, String serverURL, PlantUMLDiagramFormat format)
+            throws IOException
+    {
+        // TODO: After transition period, this default method implementation must be removed.
+        // Default method implementation is for backward compatibility only.
+        outputImage(input, output, serverURL);
+    }
```

Comment:
I think you should throw an IOException here. See the warning box at https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#HBackwardCompatibility (note that since this method is new it won't be called by old code and there's no obvious default impl we can add).